### PR TITLE
ts migrate: src/managers/collaborations.ts

### PR DESCRIPTION
**Pull request by mhagmajer**

### CODE REVIEW

File: src/managers/collaborations.ts [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+			notify?: boolean;
+			can_view_path?: boolean;
+		},
+		callback?: Function
+	) {
+		var defaultOptions = {
+			type: 'folder',
+		};
+
+		if (typeof options === 'function') {
```
Comment (swfree) [inline src/managers/collaborations.ts:175]: looks like the `options` param type def should include `| Function`

File: src/managers/collaborations.ts [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+	 */
+	createWithUserID(
+		userID: number,
+		itemID: string,
+		role: CollaborationRole,
+		options?: {
+			type?: ItemType;
+			notify?: boolean;
+			can_view_path?: boolean;
+		},
```
Comment (swfree) [inline src/managers/collaborations.ts:237]: same for here - `| Function`

File: src/managers/collaborations.ts [hunk 3]
```diff
@@ -0,0 +1,354 @@
...
+	 */
+	createWithUserEmail(
+		email: string,
+		itemID: string,
+		role: CollaborationRole,
+		options?: {
+			type?: ItemType;
+			notify?: boolean;
+			can_view_path?: boolean;
+		},
```
Comment (swfree) [inline src/managers/collaborations.ts:276]: `| Function`

File: src/managers/collaborations.ts [hunk 4]
```diff
@@ -0,0 +1,354 @@
...
+	 */
+	createWithGroupID(
+		groupID: number,
+		itemID: string,
+		role: CollaborationRole,
+		options?: {
+			type?: ItemType;
+			notify?: boolean;
+			can_view_path?: boolean;
+		},
```
Comment (swfree) [inline src/managers/collaborations.ts:315]: `| Function`

**Review by swfree — APPROVED** [review]

**Comment by mhagmajer** [comment]
I've applied your changes @swfree, thanks!

**Pull request merged**